int32 type
int32 ID
int32 value
int32 smooth
